### PR TITLE
Add support for Gradle Enterprise Gradle plugin

### PR DESCRIPTION
**Pull request by guylabs**
The project plugin and dependency versions have also been updated to the latest available.

### CODE REVIEW

File: agent/src/main/resources/build-scan-init.gradle [hunk 1]
```diff
@@ -11,12 +11,27 @@ initscript {
...
-            buildScan.buildScanPublished { scan ->
-                logger.quiet(ServiceMessage.asString(serviceMessageName, "BUILD_SCAN_URL:${scan.buildScanUri.toString()}"))
-            }
+    pluginManager.withPlugin(BUILD_SCAN_PLUGIN_ID) {
+        buildScanPublishedAction(buildScan)
+    }
+}
+
+gradle.settingsEvaluated {
+    it.pluginManager.withPlugin(GRADLE_ENTERPRISE_PLUGIN_ID) {
```
Comment (etiennestuder) [inline agent/src/main/resources/build-scan-init.gradle]: Is the `it` needed?

File: build.gradle [hunk 2]
```diff
@@ -4,15 +4,15 @@ plugins {
...
-    id 'com.github.rodm.teamcity-server' version '1.1.1' // see https://github.com/rodm/gradle-teamcity-plugin
+    id 'com.github.rodm.teamcity-server' version '1.2.2' // see https://github.com/rodm/gradle-teamcity-plugin
+    id 'com.github.rodm.teamcity-environments' version '1.2.2'
     id 'maven-publish'
     id 'com.jfrog.bintray' version '1.8.4'
 }
 
 buildScan {
     termsOfServiceUrl = 'https://gradle.com/terms-of-service'
     termsOfServiceAgree = 'yes'
```
Comment (etiennestuder) [inline build.gradle]: Please take this line out, too.

File: agent/src/main/resources/build-scan-init.gradle (hunk 1 above)
Comment (guylabs) [reply agent/src/main/resources/build-scan-init.gradle]: Yes. Not all entry points are using runtime class decoration and therefore here we still need to use the `it` prefix.

File: build.gradle (hunk 2 above)
Comment (guylabs) [reply build.gradle]: Removed

**Comment by facewindu** [comment]
@etiennestuder this one seems mergeable. The remaining question at https://github.com/etiennestuder/teamcity-build-scan-plugin/pull/10#pullrequestreview-301395738 has been answered. And validated by Luke.

**Pull request merged**